controller/utils.go

```diff
@@ -116,13 +116,21 @@ func isBackupTargetAvailable(backupTarget *longhorn.BackupTarget) bool {
 		backupTarget.Status.Available
 }
 
-// isEnginePurging returns true if the given Longhorn engine is purging.
-func isEnginePurging(engine *longhorn.Engine) bool {
-	for _, status := range engine.Status.PurgeStatus {
-		if status.IsPurging {
-			return true
-		}
+// getEngineSnapshotInfo retrieves the snapshot information for a given snapshot name
+// from the provided Longhorn engine.
+func getEngineSnapshotInfo(snapshotName string, engine *longhorn.Engine) *longhorn.SnapshotInfo {
```

Comment:
Could we rename the function to like `isSnapshotExistInEngine` and simply return `true` or `false`?